### PR TITLE
Fix object detection export notebook inference

### PR DESCRIPTION
**Pull request by yutong-xiang-97**
## What has changed and why?

This is PR 3 of a three-PR stack and is based on
`yutong-trn-2329-fix-tensorrt-gridsample`.

The object-detection export notebook used the exported model incorrectly. It depended
on the live LightlyTrain model for preprocessing information and treated the exported
outputs as already decoded labels, boxes, and scores. The exported graph actually
returns raw class logits and normalized `cxcywh` boxes, so the old notebook could use
the wrong preprocessing assumptions and decode, filter, or resize predictions
incorrectly.

The notebook now:

- reads the expected image size from the ONNX input shape;
- reads normalization values and class names from ONNX metadata;
- runs ONNX and TensorRT inference without depending on LightlyTrain;
- applies sigmoid and top-k selection before thresholding, matching `model.predict`;
- converts normalized `cxcywh` boxes to pixel-space `xyxy` boxes;
- maps internal class indices back to dataset class IDs; and
- uses TensorRT's current execution API, allocates outputs from their runtime shapes,
  and checks whether execution succeeded.

The object-detection documentation now explains that the preprocessing parameters and
class names are available directly from the exported ONNX file.

## How has it been tested?

- `make static-checks`
- End-to-end object-detection and instance-segmentation notebook runs on a Tesla T4
  with TensorRT 10.13.3.9

Raw ONNX Runtime versus TensorRT results:

- Object detection:
  - logits max/mean difference: `7.97e-4` / `1.46e-5`
  - boxes max/mean difference: `5.21e-5` / `3.58e-7`
- Instance segmentation:
  - labels: exact match
  - boxes max/mean difference: `1.05e-2` / `1.48e-4`
  - masks max/mean difference: `1.16e-3` / `2.95e-5`
  - scores max/mean difference: `6.87e-6` / `4.29e-7`

Both notebooks completed end to end and passed their raw-output parity assertions.

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (notebook and documentation update)

## Did you update the documentation?

- [x] Yes
- [ ] Not needed (internal change without effects for user)

### CODE REVIEW

File: examples/notebooks/object_detection_export.ipynb [hunk 1]
```diff
@@ -351,65 +417,52 @@
...
-    "        for name in io_names:\n",
+    "    @torch.no_grad()\n",
+    "    def __call__(self, inputs: dict[str, torch.Tensor]) -> dict[str, torch.Tensor]:\n",
+    "        buffers = {}\n",
+    "        # Bind the inputs and set their shapes before querying output shapes.\n",
+    "        for name in self.in_names:\n",
+    "            tensor = inputs[name].to(self.device).contiguous()\n",
+    "            self.context.set_input_shape(name, tuple(tensor.shape))\n",
+    "            buffers[name] = tensor\n",
+    "            self.context.set_tensor_address(name, tensor.data_ptr())\n",
```
[the remark this answers is not in the document]
Comment (yutong-xiang-97) [reply examples/notebooks/object_detection_export.ipynb]: Fixed in #916

File: examples/notebooks/object_detection_export.ipynb [hunk 2]
```diff
@@ -208,8 +203,91 @@
...
+    "    flat_index = flat_index[keep]\n",
+    "    query_index = flat_index // num_classes\n",
+    "    class_index = flat_index % num_classes\n",
+    "\n",
+    "    # Boxes are normalized to [0, 1]; scale them to the original image size.\n",
+    "    boxes = boxes_xyxy[query_index] * torch.tensor([w, h, w, h], dtype=boxes_xyxy.dtype)\n",
+    "\n",
+    "    # Map internal class indices back to the dataset class ids from the metadata.\n",
+    "    class_ids = list(classes.keys())\n",
+    "    labels = torch.tensor([class_ids[int(i)] for i in class_index])\n",
```
[the remark this answers is not in the document]
Comment (yutong-xiang-97) [reply examples/notebooks/object_detection_export.ipynb:269]: Fixed in #916

**Review by gabrielfruet — APPROVED** [review]
LGTM.

Shouldn't we export with fp16 by default in the notebook?

**Comment by yutong-xiang-97** [comment]
> Shouldn't we export with fp16 by default in the notebook?

Let's keep it as-is for all the notebooks for now.
